File: mobile/concord/lib/api/models/invite_response.dart
import 'json_utils.dart';

class InviteResponse {
  const InviteResponse({
    required this.code,
    required this.expiresAtUtc,
    required this.maxUses,
    required this.useCount,
  });

  factory InviteResponse.fromJson(Map<String, dynamic> json) {
    return InviteResponse(
      code: json.field('Code') as String,
      expiresAtUtc: parseNullableDateTime(json.field('ExpiresAtUtc')),
      maxUses: json.field('MaxUses') as int?,
      useCount: json.field('UseCount') as int,
    );
  }

  final String code;
  final DateTime? expiresAtUtc;
  final int? maxUses;
  final int useCount;
}

File: backend/Concord/Concord.Application/Models/InviteResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class InviteResponse
{
    [JsonPropertyName("Code")]
    public string? Code { get; set; }

    [JsonPropertyName("ExpiresAtUtc")]
    public DateTime? ExpiresAtUtc { get; set; }

    [JsonPropertyName("MaxUses")]
    public int? MaxUses { get; set; }

    [JsonPropertyName("UseCount")]
    public int UseCount { get; set; }
}
